cmd/erasure-multipart.go

```diff
@@ -199,38 +199,55 @@ func (er erasureObjects) cleanupStaleUploadsOnDisk(ctx context.Context, disk Sto
 	readDirFn(pathJoin(drivePath, minioMetaMultipartBucket), func(shaDir string, typ os.FileMode) error {
 		readDirFn(pathJoin(drivePath, minioMetaMultipartBucket, shaDir), func(uploadIDDir string, typ os.FileMode) error {
 			uploadIDPath := pathJoin(shaDir, uploadIDDir)
-			fi, err := disk.ReadVersion(ctx, "", minioMetaMultipartBucket, uploadIDPath, "", ReadOptions{})
-			if err != nil {
+			var modTime time.Time
+			// Upload IDs are of the form <UUID>x<UnixNano>, we can extract the time from the UUID.
+			if split := strings.Split(uploadIDDir, "x"); len(split) == 2 {
+				t, err := strconv.ParseInt(split[1], 10, 64)
+				if err == nil {
+					modTime = time.Unix(0, t)
+				}
+			}
```

Comment:
uploadID is base64 encoded string - need to decode first

Reply:
Good spot!